segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -490,6 +500,15 @@ public void mergeSegments(MergeSegments mergeSegments) {
                                                                                             properties.getLength()));
                                         });
                             return null;
+                        } else if (Exceptions.unwrap(e) instanceof BadAttributeUpdateException) {
+                            log.debug(mergeSegments.getRequestId(), "Conditional merge failed (Source segment={}, Target segment={})",
+                                    mergeSegments.getSource(), mergeSegments.getTarget(), e);
+                            final Consumer<Throwable> failureHandler = t -> {
+                                log.error("Error (Segment = '{}', Operation = '{}')", mergeSegments.getTarget(), "handling result of {}" + operation, t);
+                                connection.close();
+                            };
+                            invokeSafely(connection::send, new SegmentAttributeUpdated(mergeSegments.getRequestId(), false), failureHandler);
```

Comment:
I understand why you are using `invokeSafely`, but that is not used anywhere else in this class. If you observe how each request is handled, there's a catch-all `handleException` at the end of each future chain which should also catch problems with `connection.send`. I recommend we be consistent with regards to this behavior.

Reply:
Ok, sent response as in other parts of `PravegaRequestProcessor` (just replying back with `connection.send()`).